cloudwatch_logs_common/test/log_batch_test.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <cloudwatch_logs_common/utils/log_file_manager.h>
+
+using Aws::CloudWatchLogs::Utils::LogFileManager;
+using namespace Aws::CloudWatchLogs;
+using namespace Aws::FileManagement;
+
+const long ONE_DAY_IN_SEC = 86400000;
+
+class TestStrategy : public DataManagerStrategy {
+public:
+  bool isDataAvailable(){
+    return true;
+  }
+
+  DataToken read(std::string &data) override{
+    if(!logs.empty()){
+        data = logs.front();
+        logs.pop_front();
+    }
+
+    data_token++;
+    return data_token;
+  }
+
+  void write(const std::string &data){
+      logs.push_back(data);
+  }
+
+  void resolve(const DataToken &token, bool is_success){
+    if(is_success && token)
+        return;
+    else
+        return;
+    return;
+  }
+
+  std::list<std::string> logs;
+
+protected:
+
+  uint64_t data_token = 0;
+
+  /**
+   * Options for how and where to store files, and maximum file sizes.
+   */
+  FileManagerStrategyOptions options_;
+};
+
+class LogBatchTest : public ::testing::Test{
+public:
+  void SetUp() override {
+    test_strategy = std::make_shared<TestStrategy>();
+    file_manager = new LogFileManager(test_strategy);
+  }
+
+  void TearDown() override {
+    log_data.clear();
+    delete file_manager;
+  }
+
+  //use test_strategy to mock read/write functions from data_manager_strategy
+  std::shared_ptr<TestStrategy> test_strategy;
+  LogFileManager *file_manager;
+  LogCollection log_data;
+  Aws::CloudWatchLogs::Model::InputLogEvent input_event;
+};
+
+
+/**
+ * Test that the upload complete with CW Failure goes to a file.
+ */
+TEST_F(LogBatchTest, test_readBatch_3_of_6_pass) {
+  //add test data to logs
+  input_event.SetTimestamp(ONE_DAY_IN_SEC+1);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(2);
```

Comment:
can this be a loop? (same for below tests) - and can it be part of the fixture? Looks like it's copypasted in multiple places